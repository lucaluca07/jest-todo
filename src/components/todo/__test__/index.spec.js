import React from 'react'
import { shallow, render } from 'enzyme';
import Todo from '../index'
/**
 * Test Case
 * 1. 不传 active
 * 2. 点击禁用的 button
 * 3. 点击非禁用的 button
 * 4. 传入 active 为 completed
 */

describe('Todo', () => {
  const props = {
    onClick: jest.fn(),
    completed: false,
    text: 'use jest'
  };

  const shallowTodo = shallow(<Todo {...props} />)

  it('todo的文本为 use jest', () => {
    expect(shallowTodo.find('li').text()).toBe('use jest');
  })

  // 属性名和 react 保持一致
  it('todo的 className 为 active', () => {
    expect(shallowTodo.find('li').prop('className')).toBe('active');
  })

  it('todo的 style.textDecoration 为 ""', () => {
    expect(shallowTodo.find('li').prop('style').textDecoration).toBe('');
  })

  it('测试点击事件', () => {
    shallowTodo.find('li').at(0).simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  })

  it('改变 props 的 completed, 类名为 completed', () => {
    shallowTodo.setProps({completed: true})
    expect(shallowTodo.find('li').prop('className')).toBe('completed');
    expect(shallowTodo.find('li').prop('style').textDecoration).toBe('line-through');
  })

  it("Todo 快照测试", () => {
    const renderTodo = render(<Todo {...props} />);
    expect(renderTodo).toMatchSnapshot();
  });
})
