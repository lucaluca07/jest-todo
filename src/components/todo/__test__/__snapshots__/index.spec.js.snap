// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Todo 快照测试 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "active",
      "style": "text-decoration:",
    },
    "children": Array [
      Object {
        "data": "use jest",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "li",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
